src/metrics/include/metrics/metrics.hpp

```diff
@@ -140,13 +206,43 @@ class RegistryT final {
         return (_ops.at(actorName).at(opName)).size();
     }
 
+    void initializeMetrics(MetricsFormat format, std::string path_prefix) {
+        _format = std::move(format);
+        _path_prefix = std::move(path_prefix);
+        if (_format.use_grpc()) {
+            boost::filesystem::create_directory(_path_prefix);
+        }
+    }
```

Comment:
Yeah but since the workload context is what reads the metrics format, I had to make the registry be owned by the workload context instead of the driver, and added a method to expose a reference to it to the driver.